x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/details/index.tsx

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiLoadingSpinner, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { memo } from 'react';
+import { useParams } from 'react-router-dom';
+import { StickyContainer } from 'react-sticky';
+
+import { FiltersGlobal } from '../../../../components/filters_global';
+import { HeaderPage } from '../../../../components/header_page';
+import { DETECTION_ENGINE_PAGE_NAME } from '../../../../components/link_to/redirect_to_detection_engine';
+import { SiemSearchBar } from '../../../../components/search_bar';
+import { WrapperPage } from '../../../../components/wrapper_page';
+import { useRule } from '../../../../containers/detection_engine/rules';
+
+import {
+  indicesExistOrDataTemporarilyUnavailable,
+  WithSource,
+} from '../../../../containers/source';
+import { SpyRoute } from '../../../../utils/route/spy_routes';
+
+import { SignalsCharts } from '../../components/signals_chart';
+import { SignalsTable } from '../../components/signals';
+import { DetectionEngineEmptyPage } from '../../detection_engine_empty_page';
+import { useSignalInfo } from '../../components/signals_info';
+import { StepAboutRule } from '../components/step_about_rule';
+import { StepDefineRule } from '../components/step_define_rule';
+import { StepScheduleRule } from '../components/step_schedule_rule';
+import { buildSignalsRuleIdFilter } from '../../components/signals/default_config';
+import * as detectionI18n from '../../translations';
+import { RuleSwitch } from '../components/rule_switch';
+import { StepPanel } from '../components/step_panel';
+import { getStepsData } from '../helpers';
+import * as ruleI18n from '../translations';
+import * as i18n from './translations';
+
+export const RuleDetailsComponent = memo(() => {
+  const { ruleId } = useParams();
+  const [loading, rule] = useRule(ruleId);
+  const { aboutRuleData, defineRuleData, scheduleRuleData } = getStepsData({
+    rule,
+    detailsView: true,
+  });
+  const [lastSignals] = useSignalInfo({ ruleId });
+
+  const title = loading === true || rule === null ? <EuiLoadingSpinner size="m" /> : rule.name;
+  const subTitle =
+    loading === true || rule === null ? (
+      <EuiLoadingSpinner size="m" />
+    ) : (
+      [
+        <FormattedMessage
+          id="xpack.siem.detectionEngine.ruleDetails.ruleCreationDescription"
+          defaultMessage="Created by: {by} on 'coming soon'"
+          values={{
+            by: rule?.created_by ?? 'unknown',
```

Comment:
Should `'unknown'` be a separate translation?

Reply:
it should go bye bye